test/upgrade/main.go

```diff
@@ -336,8 +336,47 @@ func runConfigWalker(ctx context.Context, validConfigs []*configTest) {
 		time.Sleep(1 * time.Minute)
 	}
 	cancel()
-	// TODO: Replace sleep with wait for the existing healthy Game Servers finish naturally by reaching their shutdown phase.
-	time.Sleep(30 * time.Second)
+
```

Comment:
Whoops! That one fell through the cracks. Glad to the TODO was there to provide guideance.

LGTM, but would like @igooch to triple check as she was the one who originally wrote this.